### ISSUE TITLE
Add One shot systems

### ISSUE THREAD

**Issue by delaneyj** [mlange-42/arche-model#55]
Be nice for setup to have some systems run initialize then remove self

**Comment by mlange-42** [comment]
Hi @delaneyj!

thanks for coming along!

You can easily do this yourself, as the systems are accessible as the resource `Systems`.
About like this (not tested):

```go
func (s *MySys) Initialize(world *ecs.World) {
    // Do your stuff...

    sysRes := generic.NewResource[model.Systems](world)
    systems := sysRes.Get()
    systems.RemoveSystem(s)
}
```

**Comment by delaneyj** [comment]
Right, I just thought it might be clearer to have a OneShot intereface that only has Init and a System interface extends that interface to be more explicit and to auto handle to removal.

**Comment by mlange-42** [comment]
Well, ok, this is worth a consideration.
